pkg/pkgmiddleware/requestId.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (C) 2022 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package pkgmiddleware
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/spacemonkeygo/monkit/v3"
+)
+
+// Key to use when setting the request ID.
+type ctxKeyRequestID string
```

Comment:
nit: a more robust context key would be something like this:

```suggestion
type RequestIDKey struct{}
```

then you can just:

```
... context.WithValue(ctx, RequestIDKey{}, ...) ...
```

and then:

```
... ctx.Value(RequestIDKey{}).(...) ...
```